mod/item.php

```diff
@@ -557,39 +553,43 @@ function item_post(App $a) {
 		$toplevel_parent = q("SELECT `contact`.* FROM `contact`
 						INNER JOIN `item` ON `item`.`contact-id` = `contact`.`id` AND `contact`.`url` = `item`.`author-link`
 						WHERE `item`.`id` = `item`.`parent` AND `item`.`parent` = %d", intval($parent));
-		if ($toplevel_parent)
-			$toplevel_contact = '@'.$toplevel_parent[0]['nick'].'+'.$toplevel_parent[0]['id'];
-		else {
+		if (dbm::is_result($toplevel_parent)) {
+			$toplevel_contact = '@' . $toplevel_parent[0]['nick'] . '+' . $toplevel_parent[0]['id'];
+		} else {
 			$toplevel_parent = q("SELECT `author-link`, `author-name` FROM `item` WHERE `id` = `parent` AND `parent` = %d", intval($parent));
 			$toplevel_contact = '@[url='.$toplevel_parent[0]['author-link'].']'.$toplevel_parent[0]['author-name'].'[/url]';
 		}
 
-		if (!in_array($toplevel_contact,$tags))
+		if (!in_array($toplevel_contact, $tags)) {
 			$tags[] = $toplevel_contact;
+		}
 	}
 
 	$tagged = array();
 
 	$private_forum = false;
 
-	if(count($tags)) {
-		foreach($tags as $tag) {
+	if (count($tags)) {
+		foreach ($tags as $tag) {
 
-			if(strpos($tag,'#') === 0)
+			if (strpos($tag, '#') === 0) {
 				continue;
+			}
 
-			// If we already tagged 'Robert Johnson', don't try and tag 'Robert'.
-			// Robert Johnson should be first in the $tags array
-
+			/*
+			 * If we already tagged 'Robert Johnson', don't try and tag 'Robert'.
+			 * Robert Johnson should be first in the $tags array
+			 */
 			$fullnametagged = false;
-			for($x = 0; $x < count($tagged); $x ++) {
-				if(stristr($tagged[$x],$tag . ' ')) {
+			for ($x = 0; $x < count($tagged); $x ++) {
+				if (stristr($tagged[$x],$tag . ' ')) {
```

Comment:
`$tagged` is being initialized ~10 lines above but not written to before that for() (now foreach) loop is being reached. maybe old-lost code here?